data/common.yaml

```diff
@@ -0,0 +1,129 @@
+---
+lookup_options:
+  simp_snmpd::v3_users_hash:
+    merge:
+      strategy: deep
+      knockout_prefix: --
+  simp_snmpd::view_hash:
+    merge:
+      strategy: deep
+      knockout_prefix: --
+  simp_snmpd::group_hash:
+    merge:
+      strategy: deep
+      knockout_prefix: --
+  simp_snmpd::group_hash:
+    merge:
+      strategy: deep
+      knockout_prefix: --
+  simp_snmpd::listeningaddr:
+    merge:
+      strategy: deep
+      knockout_prefix: --
+
+# Defaults for creation of users, groups views and access
+#  If not specifically set these will be used.  They are
+#  also used for connections.
+simp_snmpd::defauthtype: 'SHA'
+simp_snmpd::defprivtype: 'AES'
+simp_snmpd::defsecuritymodel: 'usm'
+simp_snmpd::defsecuritylevel: 'priv'
+#V3 users, groups, views and access entrys for the snmpd.conf
+#Set password to undefined to have simp autogenerate it.
+simp_snmpd::v3_users_hash:
```

Comment:
If this data structure or any data structure needs more documentation, it should be in the init.pp or in it's own section in the README. Users wouldn't read this file to understand the data.